consensus/src/consensus_observer/observer.rs

```diff
@@ -805,6 +725,29 @@ impl ConsensusObserver {
         }
     }
 
+    /// Produces a list of sorted peers to service our subscription request. Peers
+    /// are prioritized by validator distance and latency. If `previous_subscription_peer`
+    /// is provided, it will be excluded from the selection process.
+    fn sort_peers_for_subscription(
+        &mut self,
+        previous_subscription_peer: Option<PeerNetworkId>,
+    ) -> Option<Vec<PeerNetworkId>> {
+        if let Some(mut peers_and_metadata) = self.get_connected_peers_and_metadata() {
```

Comment:
I remember @ibalajiarun figured that cloning the HashMap is quite expensive (>10ms), we probably should use reference if possible

Reply:
Yeah, makes sense. Thankfully we're only doing this (at most) every 5 seconds, so we should be good. But, I'll see if I can clean it up in the future. 😄